notebook/converted/benchmark_results.py

```diff
@@ -0,0 +1,271 @@
+plot += ggtitle('PCA vs untransformed data for classifying sepsis')
+print(plot)
+
+
+# ## All labels
+
+# In[6]:
+
+
+in_files = glob.glob('../../results/all_labels.*')
+print(in_files[:5])
+
+
+# In[7]:
+
+
+metrics = None
+for path in in_files:
+    if metrics is None:
+        metrics = pd.read_csv(path, sep='\t')
+        
+        model_info = path.strip('.tsv').split('all_labels.')[-1]
+        
+        model_info = model_info.split('.')
+        
+        if len(model_info) == 4:
+            unsupervised_model = model_info[0]
+            supervised_model = model_info[1]
+        else:
+            unsupervised_model = 'untransformed'
+            supervised_model = model_info[0]
+             
+        metrics['unsupervised'] = unsupervised_model
+        metrics['supervised'] = supervised_model
+    else:
+        new_df = pd.read_csv(path, sep='\t')
+        model_info = path.strip('.tsv').split('all_labels.')[-1]
+        model_info = model_info.split('.')
+        
+        if len(model_info) == 4:
+            unsupervised_model = model_info[0]
+            supervised_model = model_info[1]
+        else:
+            unsupervised_model = 'untransformed'
+            supervised_model = model_info[0]
+             
+        new_df['unsupervised'] = unsupervised_model
+        new_df['supervised'] = supervised_model
+        
+        metrics = pd.concat([metrics, new_df])
+
+metrics
+
+
+# In[8]:
+
+
+plot = ggplot(metrics, aes(x='supervised', y='accuracy', fill='unsupervised')) 
+plot += geom_violin()
+plot += ggtitle('PCA vs untransformed data for all label classification')
+print(plot)
+
+
+# In[9]:
+
+
+plot = ggplot(metrics, aes(x='supervised', y='accuracy', fill='unsupervised')) 
+plot += geom_jitter(size=2)
+plot += ggtitle('PCA vs untransformed data for all label classification')
+print(plot)
+
+
+# # Subsets of healthy labels
+
+# In[10]:
+
+
+in_files = glob.glob('../../results/subset_label.sepsis*')
+print(in_files[:5])
+
+
+# In[11]:
+
+
+sepsis_metrics = pd.DataFrame()
+for path in in_files:
+    new_df = pd.read_csv(path, sep='\t')
+    model_info = path.strip('.tsv').split('sepsis.')[-1]
+    model_info = model_info.split('.')
+        
+    if len(model_info) == 4:
+        unsupervised_model = model_info[0]
+        supervised_model = model_info[1]
+    else:
+        unsupervised_model = 'untransformed'
+        supervised_model = model_info[0]
+             
+    new_df['unsupervised'] = unsupervised_model
+    new_df['supervised'] = supervised_model
+        
+    sepsis_metrics = pd.concat([sepsis_metrics, new_df])
+    
+sepsis_metrics = sepsis_metrics.rename({'fraction of healthy used': 'healthy_used'}, axis='columns')
+sepsis_metrics['healthy_used'] = sepsis_metrics['healthy_used'].round(1)
+    
+sepsis_metrics
+
+
+# In[12]:
+
+
+print(sepsis_metrics[sepsis_metrics['healthy_used'] == 1])
+
+
+# In[13]:
+
+
+plot = ggplot(sepsis_metrics, aes(x='factor(healthy_used)', y='accuracy', )) 
```

Comment:
The test sets are subsetted the same amount as the training sets, unless doing so would leave you with 0 samples. I did so to prevent the performance from being confounded by the label distribution between the training and test sets